scripts/genCurrenciesNumber.mjs

```diff
@@ -0,0 +1,30 @@
+/* eslint-disable functional/no-expression-statement */
+import path from 'path';
+import * as url from 'url';
+
+import { genCurrencies } from './genCurrencies.mjs';
+
+/**
+ * Script to generate Currency<number> source files.
+ */
+genCurrencies(
+  // Output directory
+  path.resolve(
+    url.fileURLToPath(new URL('.', import.meta.url)), // __dirname es module style
+    '../packages/currencies/src/iso4217/amendments/168'
+  ),
+  // Source format function
+  ({ base, code, description, exponent }) =>
+    `
+      import type { Currency } from '../../../types';
+
+      /**
+       * ${description}.
+       */
+      export const ${code}: Currency<number> = {
```

Comment:
This could easily be adopted to be even more generic (abstractions all the way!), By letting the `number` part (and the formatting of the numbers down below), be configured.

This is just a pop-up thought, and might just be a bit too much abstraction layering though.

Reply:
@reckter checkout out the abstraction now. Creating the currency generation config adds a `mapNumber` function which converts the JS number type of the json data file into a string that will be output into the generated source files.

Example for `bigint`

```ts
createCurrencyConfig({
  genericType: 'bigint',
  mapNumber: (n) => `${n}n`,
}
```